File: wisefy/src/androidTest/java/com/isupatches/wisefy/main/IsNetworkWPATests.java
package com.isupatches.wisefy.main;

import static org.junit.Assert.assertEquals;

import android.net.wifi.ScanResult;

import com.isupatches.wisefy.AbstractBaseAndroidJUnit4TestClass;
import com.isupatches.wisefy.GeneratorUtil;
import com.isupatches.wisefy.WiseFy;
import com.isupatches.wisefy.constants.Capabilities;

import org.junit.Test;

/**
 * Tests the ability to determine if a network has WPA security.
 *
 * @see WiseFy#isNetworkWPA(ScanResult)
 *
 * @author Patches
 */
public class IsNetworkWPATests extends AbstractBaseAndroidJUnit4TestClass {

  public IsNetworkWPATests() {
    // No-op
  }

  @Test
  public void failure_differentCapability() {
    final ScanResult scanResult = GeneratorUtil.createMockAccessPointWithCapabilities("Other");
    assertEquals(false, getWiseFy().isNetworkWPA(scanResult));
  }

  @Test
  public void failure_emptyCapabilities() {
    final ScanResult scanResult = GeneratorUtil.createMockAccessPointWithCapabilities("");
    assertEquals(false, getWiseFy().isNetworkWPA(scanResult));
  }

  @Test
  public void failure_nullCapabilities() {
    final ScanResult scanResult = GeneratorUtil.createMockAccessPointWithCapabilities(null);
    assertEquals(false, getWiseFy().isNetworkWPA(scanResult));
  }

  @Test
  public void failure_nullScanResult() {
    assertEquals(false, getWiseFy().isNetworkWPA(null));
  }

  @Test
  public void success() {
    final ScanResult scanResult = GeneratorUtil.createMockAccessPointWithCapabilities(Capabilities.WPA);
    assertEquals(true, getWiseFy().isNetworkWPA(scanResult));
  }
}

File: wisefy/src/androidTest/java/com/isupatches/wisefy/main/IsNetworkWPATests.kt
package com.isupatches.wisefy.main

import com.isupatches.wisefy.BaseAndroidJUnit4TestClass
import com.isupatches.wisefy.constants.WPA
import com.isupatches.wisefy.createMockAccessPointWithCapabilities

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Tests the ability to determine if a network has WPA security.
 *
 * @author Patches
 */
internal class IsNetworkWPATests : BaseAndroidJUnit4TestClass() {

    @Test fun failure_differentCapability() {
        val scanResult = createMockAccessPointWithCapabilities("Other")
        assertEquals(false, wiseFy.isNetworkWPA(scanResult))
    }

    @Test fun failure_emptyCapabilities() {
        val scanResult = createMockAccessPointWithCapabilities("")
        assertEquals(false, wiseFy.isNetworkWPA(scanResult))
    }

    @Test fun failure_nullCapabilities() {
        val scanResult = createMockAccessPointWithCapabilities(null)
        assertEquals(false, wiseFy.isNetworkWPA(scanResult))
    }

    @Test fun failure_nullScanResult() {
        assertEquals(false, wiseFy.isNetworkWPA(null))
    }

    @Test fun success() {
        val scanResult = createMockAccessPointWithCapabilities(WPA)
        assertEquals(true, wiseFy.isNetworkWPA(scanResult))
    }
}
